python/shotgun_model/shotgun_model.py

```diff
@@ -1291,7 +1374,10 @@ def __realize_parent_child_hierarchy_r(self, node):
         # children are always sorted alphabetically underneath their parent.
         for child_key in sorted(children.keys(), cmp=lambda x,y: cmp(x.lower(), y.lower())):
             child = children[child_key]
-            node["item"].appendRow(child["item"])
+
+            on_leaf_level = not bool(child.get("children"))
```

Comment:
Had that, but ran into a case where children was in child but was an empty dict.  So switched to this.